packages/@lwc/engine-core/src/framework/reporting.ts

```diff
@@ -85,10 +118,10 @@ export function onReportingEnabled(callback: OnReportingEnabledCallback) {
 /**
  * Report to the current dispatcher, if there is one.
  * @param reportingEventId
- * @param vm
+ * @param payload - data to report
  */
-export function report(reportingEventId: ReportingEventId, vm?: VM) {
+export function report<T extends ReportingEventId>(reportingEventId: T, payload: Payload<T>) {
     if (enabled) {
-        currentDispatcher(reportingEventId, vm?.tagName, vm?.idx);
+        currentDispatcher(reportingEventId, payload);
```

Comment:
I don't think `idx` is very useful, so I got rid of it. As for `tagName`, it turned out to be way simpler for the reporter to just pass in the `tagName` if they already have the `vm`.

Reply:
Good move. Was just thinking about this a little and I'm not sure what we would do with knowing the VM index if we were looking at a log/metric.